From: Cyrill Gorcunov <gorcunov@openvz.org>
Date: Tue, 3 Feb 2009 19:45:36 +0000 (+0300)
Subject: Introduce new "ppp" feature
X-Git-Url: http://git.openvz.org/?p=vzctl;a=commitdiff_plain;h=d78a7653d983269553763d536f69de0bb5109353

Introduce new "ppp" feature

Allow an user to turn "ppp" feature on and off.

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
---

diff --git a/include/linux/vzcalluser.h b/include/linux/vzcalluser.h
index 045a0e1..66d61cd 100644
--- a/include/linux/vzcalluser.h
+++ b/include/linux/vzcalluser.h
@@ -202,6 +202,7 @@ struct env_create_param3 {
 #define VE_FEATURE_DEF_PERMS	(1ULL << 2)
 #define VE_FEATURE_SIT		(1ULL << 3)
 #define VE_FEATURE_IPIP		(1ULL << 4)
+#define VE_FEATURE_PPP		(1ULL << 5)
 
 #define VE_FEATURES_OLD		(VE_FEATURE_SYSFS)
 #define VE_FEATURES_DEF		(VE_FEATURE_SYSFS | \
diff --git a/man/vzctl.8 b/man/vzctl.8
index 85d3b0d..92d49e3 100644
--- a/man/vzctl.8
+++ b/man/vzctl.8
@@ -446,7 +446,7 @@ designated by its \fImajor\fR and \fIminor\fR numbers. Device file have to be cr
 .TP
 \fB--features\fR \fIname\fR:\fBon|off\fR
 Enable or disable a specific container feature.
-Known features are: \fIsysfs\fR, \fInfs\fR, \fIsit\fR, \fIipip\fR.
+Known features are: \fIsysfs\fR, \fInfs\fR, \fIsit\fR, \fIipip\fR, \fIppp\fR.
 .TP
 \fBApply config\fR
 .TP
diff --git a/src/lib/vzfeatures.c b/src/lib/vzfeatures.c
index 2f5596a..c6af929 100644
--- a/src/lib/vzfeatures.c
+++ b/src/lib/vzfeatures.c
@@ -27,6 +27,7 @@ static struct feature_s features[] = {
 	{ "nfs",	0,	VE_FEATURE_NFS },
 	{ "sit",	0,	VE_FEATURE_SIT},
 	{ "ipip",	0,	VE_FEATURE_IPIP},
+	{ "ppp",	0,	VE_FEATURE_PPP},
 	{ NULL}
 };
 
